pandas/core/generic.py

```diff
@@ -7715,34 +7715,107 @@ def last(self, offset):
     def rank(self, axis=0, method='average', numeric_only=None,
              na_option='keep', ascending=True, pct=False):
         """
-        Compute numerical data ranks (1 through n) along axis. Equal values are
-        assigned a rank that is the average of the ranks of those values
+        Compute numerical data ranks (1 through n) along axis.
+
+        By default, equal values are assigned a rank that is the average of the
+        ranks of those values.
 
         Parameters
         ----------
         axis : {0 or 'index', 1 or 'columns'}, default 0
-            index to direct ranking
-        method : {'average', 'min', 'max', 'first', 'dense'}
-            * average: average rank of group
-            * min: lowest rank in group
-            * max: highest rank in group
-            * first: ranks assigned in order they appear in the array
-            * dense: like 'min', but rank always increases by 1 between groups
-        numeric_only : boolean, default None
+            Index to direct ranking.
+        method : {'average', 'min', 'max', 'first', 'dense'}, default 'average'
+            Which method to use to rank equal values:
+            * average: average rank of group.
+            * min: lowest rank in group.
+            * max: highest rank in group.
+            * first: ranks assigned in order they appear in the array.
+            * dense: like 'min', but rank always increases by 1 between groups.
+        numeric_only : bool, default None
             Include only float, int, boolean data. Valid only for DataFrame or
-            Panel objects
-        na_option : {'keep', 'top', 'bottom'}
-            * keep: leave NA values where they are
-            * top: smallest rank if ascending
-            * bottom: smallest rank if descending
-        ascending : boolean, default True
-            False for ranks by high (1) to low (N)
-        pct : boolean, default False
-            Computes percentage rank of data
+            Panel objects.
+        na_option : {'keep', 'top', 'bottom'}, default 'keep'
+            How to rank NaN values:
+            * keep: assign NaN rank to NaN values.
+            * top: assign smallest rank to NaN values if ascending.
+            * bottom: assign highest rank to NaN values if ascending.
+        ascending : bool, default True
+            False for ranks by high (1) to low (N).
+        pct : bool, default False
+            Computes percentage rank of data.
 
         Returns
         -------
-        ranks : same type as caller
+        ranks : Series or DataFrame
+
+        Examples
+        --------
+
+        The default behavior returns average ranks of every columns
+
+        >>> df = pd.DataFrame(data={'Customer':['A','B','C','D','E'],
```

Comment:
Thank you for linking this.

Do you think I should also remove the example `df.rank(numeric_only=True)` ? Since rank should only work with numerical columns in future releases, I assume the example is redundant.

Reply:
Yep good idea and that makes a lot of sense

Reply:
Some spaces are missing to be pep8 valid.